guides/source/caching_with_rails.md

```diff
@@ -370,6 +370,18 @@ There are some common options that can be used by all cache implementations. The
 
 * `:race_condition_ttl` - This option is used in conjunction with the `:expires_in` option. It will prevent race conditions when cache entries expire by preventing multiple processes from simultaneously regenerating the same entry (also known as the dog pile effect). This option sets the number of seconds that an expired entry can be reused while a new value is being regenerated. It's a good practice to set this value if you use the `:expires_in` option.
 
+When using Puma or another threaded server it is recommended to use a connection pool to prevent thread contention. Connection pooling is only supported by the `MemCacheStore` and `RedisCacheStore`. To enable the connection pool first add the `connection_pool` gem to your Gemfile.
+
+```ruby
+gem 'connection_pool'
+```
+
+Next set either the `:pool_size` and/or `:pool_timeout` options.
+
+* `:pool_size` - This options sets the number of connections per process. It is recommended to set it to the number of threads per process of the app server.
+
+* `:pool_timeout` - This option sets the amount of time in seconds after which a connection times out. If no connection is available within timeout seconds it will raise a Timeout::Error.
```

Comment:
The first sentence currently sounds like it describes an idle timeout, rather than a checkout timeout.

```suggestion
* `:pool_timeout` - This option sets the amount of time in seconds to wait for a connection. If no connection is available within the timeout, a `Timeout::Error` will be raised.
```

Reply:
Good point! 🙂 I Actually thought that was how it works.
I've made the suggested changes (and added the default values as well).